On-death thought rework

Snowstar38 opened the pull request:
- Death thoughts have their own .json and are now no longer in cats.py
- Now others can introduce additional variants for each type of death thought
- Also adds pronoun tagging for ondeath thoughts as this was not possible before

Snowstar38 committed:
Death thought rework

Creates a new type of thought, "death", that is applied when a cat dies

- Death thoughts have their own .json and are now no longer in cats.py
- Now others can introduce additional variants for each type of death thought

Snowstar38 committed:
Changed from "death" to "ondeath" for clarity

scribblecrumb commented:
Leader death thoughts don't seem to be working.  I tried having my leader die both naturally via timeskip and manually via profile, either one gave a death thought.  Regular cats do seem to work fine, though.

scribblecrumb reviewed (changes requested):
Leader death thoughts don't seem to be working. I tried having my leader die both naturally via timeskip and manually via profile, neither one gave a death thought. Regular cats do seem to work fine, though.

Snowstar38 commented:
Ah, I just saw this. When I get home I will take a look and try to fix this. Thanks for telling me!

Snowstar38 commented:
@scribblecrumb Hey, I apologize for taking so long to look into this - life stuff. But I just tried to do this and leader thoughts work fine for me? I cloned the dev branch and merged in only these file changes, but my leader death thoughts appear properly for me under both circumstances 🤔

scribblecrumb commented:
Hmmm very odd, this is the error I get when my leader dies.

![image](https://github.com/ClanGenOfficial/clangen/assets/115410010/e1fb32f7-4409-422e-8119-6c388ad2ba7a)

It looks like your branch is pretty far behind dev rn too, you may want to sync it.  Perhaps that's making a difference in what I'm experiencing?

![image](https://github.com/ClanGenOfficial/clangen/assets/115410010/21ae3c7f-bed8-4c1f-81aa-784fd36b165e)

Snowstar38 committed:
Merge branch 'development' into Death-thoughts

Snowstar38 committed:
Fixing it up!

Snowstar38 commented:
@scribblecrumb 🤦‍♀️ I have no idea how I managed to not notice I hadn't updated the branch, haha!

In any case, once I updated it, I ran into the same issues you'd mentioned. I think they should be fixed now. Thanks for being patient with me about this 😅

Snowstar38 committed:
fix crash

Snowstar38 committed:
Merge branch 'ClanGenOfficial:development' into Death-thoughts

Snowstar38 committed:
crash fix, please stop reverting commits

scribblecrumb commented on scripts/cat/cats.py:
```diff
@@ -558,12 +563,10 @@ def die(self, body: bool = True):
-                "Is surprised to find themself walking the stars of Silverpelt"
-            )
+            death_thought = Thoughts.new_death_thought(self, darkforest, isoutside)
+            final_thought = event_text_adjust(self, death_thought, self)
```
make sure that you specify `main_cat=self` in the event_text_adjust arguments, otherwise it'll think `self` is the patrol leader, not the main cat.

scribblecrumb commented on scripts/cat/thoughts.py:
```diff
@@ -330,3 +330,57 @@ def get_chosen_thought(main_cat, other_cat, game_mode, biome, season, camp):
+            created_list.append(inter)
+        return created_list
+    
+    def leader_death_thought(self, lives_left, darkforest):
```
lets add some docstrings to these new functions!  Make sure to include the params with `:param name_of_param: explanation of param`

scribblecrumb commented on scripts/utility.py:
```diff
@@ -1894,7 +1894,10 @@ def event_text_adjust(
+        if main_cat:
+            replace_dict["m_c"] = (str(main_cat.name), choice(main_cat.pronouns))
+        else:
+            replace_dict["m_c"] = (str(Cat.name), choice(Cat.pronouns))
```
I don't believe this quite works as intended (nor do I think it's a necessary change, you just need to pass the cat as `main_cat`) `Cat` is meant to be the Cat class, not a cat object.  So it looks like you're calling for the Cat class `.name` and `.pronouns`, not a cat object's `.name` and `.pronouns`

Snowstar38 committed:
All hail scribble

- Added main_cat=self to event_text_adjust
- Removed utility.py edit as above change rendered it unnecessary
- Added parameter explanations to leader_death_thought

Snowstar38 commented:
@scribblecrumb I have followed your suggestions! The edit to utility.py was just because I didn't know why it kept crashing aha; you figured out it was the main_cat=self part so now it's unnecessary and I removed it. :)
